pymc3/distributions/multivariate.py

```diff
@@ -2098,3 +2098,62 @@ def logp(value, mu, W, alpha, tau):
             at.all(alpha >= -1),
             tau > 0,
         )
+
+
+class StickBreakingWeightsRV(RandomVariable):
+    name = "stick_breaking_weights"
+    ndim_supp = 1
+    ndims_params = [0]
+    dtype = "floatX"
+    _print_name = ("StickBreakingWeights", "\\operatorname{StickBreakingWeights}")
+
+    def __call__(self, alpha, size=None, **kwargs):
+        return super().__call__(alpha, size=size, **kwargs)
+
+    def _shape_from_params(self, dist_params, rep_param_idx=1, param_shapes=None):
+        return default_shape_from_params(self.ndim_supp, dist_params, rep_param_idx, param_shapes)
```

Comment:
This was temporarily taken from [`DirichletMultinomialRV`](https://github.com/pymc-devs/pymc3/blob/7e464e59bcb0adb28df94f379b3e8d4af12bd4d1/pymc3/distributions/multivariate.py#L533-L534), but it still isn't correct.

Reply:
`rep_param_idx=1` is incorrect since your distribution only has 1 argument (zero based indexing). Isn't the shape of your distribution given by `return dist_params`?

Reply:
Hmm perhaps, but frankly I don't fully understand what `_shape_from_params` is trying to achieve and the difference between `ndim_supp` and the output shape of `RandomVariable`s.

I added a parameter `K` that determines the number of sticks to break and the length of the sampled vector.

Using `return dist_params` works but it seems to require `alpha` to be discrete as yields the following error displayed in the notebook [here](https://github.com/larryshamalama/pymc3-playground/blob/master/notebooks/progress/stickbreakingweights.ipynb):

```{python}
~/anaconda3/envs/pymc3-dev-py38/lib/python3.8/site-packages/aesara/tensor/basic_opt.py in on_import(self, fgraph, node, reason)
   1334                 # keep it this way. See #266 for a better long-term fix.
   1335                 if getattr(d, "dtype", "int64") != "int64":
-> 1336                     assert d.dtype in discrete_dtypes, (node, d.dtype)
   1337                     assert str(d.dtype) != "uint64", node
   1338                     new_shape += sh[len(new_shape) : i + 1]

AssertionError: (stick_breaking_weights_rv{1, (0, 0), floatX, False}(RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FDE47636120>), TensorConstant{[]}, TensorConstant{11}, TensorConstant{2.0}, TensorConstant{5}), 'float32')
```

Reply:
`ndim_supp` provides information about the minimum number of dimensions of the basic distribution. For example a beta distribution is a vector.

But this, by itself, is not sufficient to know the output shape of a specific beta variable.

`_shape_from_params` does this. Given the parameters and/or their shapes, what is the output shape?

For example, the beta shape is fully determined by the size of the alpha parameter, so the method could return  `alpha.shape`.

By default, `_shape_from_params` tries to make an educated guess, but it does not always work, and that's when you need to completely override the default method (or, sometimes, just nudge it slightly as in the MultinomialRV case)

Reply:
Thanks for the explanation above!

Not sure if this is reflected as default in the code, I added a commit that adds `K` as a parameter.

(Thinking out loud) Using a scalar `alpha`, the output vector would have a shape of `(K + 1,)`. Then, would the shape be `alpha.shape + (K + 1,)` or something like this? I'm not exactly sure how we would write this in Aesara